cmd/pinniped/cmd/doc.go

```diff
@@ -0,0 +1,69 @@
+package cmd
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/cobra/doc"
+)
+
+// GenMarkdown generates markdown documentation for the Pinniped CLI at the file at the provided
+// path.
+//
+// If GenMarkdown encouters an error, it will print the error and call os.Exit(1).
+func GenMarkdown(path string) {
+	var generated bytes.Buffer
+	generate(&generated)
+	writeToPath(path, &generated)
+}
+
+func generate(w io.Writer) {
+	generateHeader(w)
+	generateCommand(w, rootCmd)
+}
+
+func generateHeader(w io.Writer) {
+	_, err := fmt.Fprintf(w, `---
+title: Command-Line Options Reference
+description: Reference for the `+"`pinniped`"+` command-line tool
+cascade:
+  layout: docs
+menu:
+  docs:
+    name: Command-Line Options
+    weight: 30
+    parent: reference
+---
+
+`)
+	maybeDie(err)
+}
+
+func generateCommand(w io.Writer, command *cobra.Command) {
+	for _, command := range command.Commands() {
+		// generate children
+		generateCommand(w, command)
+
+		// generate self
+		if !command.Hidden && (command.Run != nil || command.RunE != nil) {
```

Comment:
I think this logic is not quite right yet, because the non-hidden commands under a hidden parent will still be documented (the `pinniped login [...]` subcommands are like this).

We could probably also just mark those children as also hidden.

Reply:
Mmmm, OK. I'll update the traversal to not visit a subtree when the parent is hidden.

Reply:
022d91549514b3c64d2ad52d5766eb07d1099c72